src/test/scala/mesosphere/marathon/integration/ReadinessCheckIntegrationTest.scala

```diff
@@ -40,35 +41,38 @@ class ReadinessCheckIntegrationTest extends AkkaIntegrationFunTest with Embedded
     deploy(serviceProxy("/noreadynohealth".toTestPath, "phase()", withHealth = false), continue = false)
   }
 
-  test("An upgrade of an application will wait for the readiness checks", Unstable) {
+  test("An upgrade of an application will wait for the readiness checks") {
     val serviceDef = serviceProxy("/upgrade".toTestPath, "phase(block1!,block2!,block3!)", withHealth = false)
     deploy(serviceDef, continue = true)
 
-    When("The service is upgraded")
-    val oldTask = marathon.tasks(serviceDef.id).value.head
-    val update = marathon.updateApp(serviceDef.id, AppUpdate(env = Some(EnvVarValue(sys.env))))
-    val newTask = eventually {
-      marathon.tasks(serviceDef.id).value.find(_.id != oldTask.id).get
-    }
+      When("The service is upgraded")
+      val oldTask = marathon.tasks(serviceDef.id).value.head
+      val update = marathon.updateApp(serviceDef.id, AppUpdate(env = Some(EnvVarValue(sys.env))))
+
+        And("The ServiceMock is up")
+      val serviceFacade = createServiceFacade(serviceDef.id){ task =>
+        task.id != oldTask.id&& task.state == MesosProtos.TaskState.TASK_RUNNING.name()
```

Comment:
Oh thanks for catching. I've changed that line again in a follow-up with related RestartIntegrationTest (which also uses ServiceMock so I've changed that to be a little more generic) and it's fixed there :)